### PR TITLE
Adds a simple and straight forward wrapper for the qdrant API

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/java/sirius/db/qdrant/Qdrant.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import sirius.kernel.commons.Strings;
+import sirius.kernel.di.std.ConfigValue;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.health.Average;
+import sirius.kernel.health.Log;
+import sirius.kernel.health.metrics.MetricProvider;
+import sirius.kernel.health.metrics.MetricsCollector;
+
+/**
+ * Provides access to a qdrant vector dataase.
```
Comment (sabieber) [inline src/main/java/sirius/db/qdrant/Qdrant.java:20]:
```suggestion
 * Provides access to a qdrant vector database.
```

File: src/main/java/sirius/db/qdrant/Point.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    public static String deriveId(String identifier) {
+        return Hasher.sha1().hash(identifier).toHexString();
+    }
+
+    /**
+     * Creates a new point with the given vector.
+     *
+     * @param id     the id of the point. Note that this should be a UUID or at least a string of 32 bytes which only
+     *               use hexadecimal characters (0-F). User either {@link UUID} or {@link #deriveId(String)}.
+     * @param vector the vector to store. User {@link Tensors} to decode string or list representations. Note that the
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/qdrant/Point.java:43]:
```suggestion
     * @param vector the vector to store. Use {@link Tensors} to decode string or list representations. Note that the
```

File: src/main/java/sirius/db/qdrant/QdrantDatabase.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+ */
+public class QdrantDatabase {
+
+    private static final String QDRANT_DEFAULT_HOST = "qdrant";
+    private static final int QDRANT_DEFAULT_PORT = 6333;
+    private static final Duration CONNECT_TIMEOUT = Duration.ofSeconds(5);
+    private static final Duration DEFAULT_OPERATION_TIMEOUT = Duration.ofSeconds(60);
+    private static final String HEADER_CONTENT_TYPE = "Content-Type";
+    private static final String CONTENT_TYPE_APPLICATION_JSON = "application/json";
+    private static final int HTTP_STATUS_OK = 200;
```
Comment (sabieber) [inline src/main/java/sirius/db/qdrant/QdrantDatabase.java:40]: Aren't there common constants available for these?

File: src/main/java/sirius/db/qdrant/QdrantDatabase.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        Similarity(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    enum Method {
```
Comment (sabieber) [inline src/main/java/sirius/db/qdrant/QdrantDatabase.java:60]: No access modifier? And isn't there a common enum or constants present for these?

File: src/main/java/sirius/db/qdrant/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# qdrant Database API
+
+Provides a simple API to access the [qdrant](https://qdrant.tech/) database.
+
+Qdrant is a fast and scalable vector search engine. A common AI task is to convert text or images into vectors
+using embeddings. These vectors are then stored in a database. During retrieval either an example text, image or
+query is also transformed into a vector and the task of the vector database is to find nearest neighbors in a
+quick and efficient manner.
+
+The implementation is lighwieght and does not require any external dependencies as all requests are sent via
```
Comment (sabieber) [inline src/main/java/sirius/db/qdrant/README.md:10]:
```suggestion
The implementation is lightweight and does not require any external dependencies as all requests are sent via
```

File: src/main/java/sirius/db/qdrant/Search.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    private List<Tuple<String, Object>> mustNotFilters;
+
+    protected Search(QdrantDatabase qdrantDatabase, String collection, float[] vector) {
+        this.qdrantDatabase = qdrantDatabase;
+        this.collection = collection;
+        this.vector = vector.clone();
+    }
+
+    /**
+     * Sepcifies a payload filter which must match.
```
Comment (sabieber) [inline src/main/java/sirius/db/qdrant/Search.java:36]:
```suggestion
     * Specifies a payload filter which must match.
```

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/db/qdrant/QdrantDatabase.java [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+public class QdrantDatabase {
+
+    private static final String QDRANT_DEFAULT_HOST = "qdrant";
+    private static final int QDRANT_DEFAULT_PORT = 6333;
+    private static final Duration CONNECT_TIMEOUT = Duration.ofSeconds(5);
+    private static final Duration DEFAULT_OPERATION_TIMEOUT = Duration.ofSeconds(60);
+    private static final String HEADER_CONTENT_TYPE = "Content-Type";
+    private static final String CONTENT_TYPE_APPLICATION_JSON = "application/json";
+    private static final int HTTP_STATUS_OK = 200;
+    private static final String URI_PREFIX_COLLECTIONS = "/collections/";
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/qdrant/QdrantDatabase.java:41]: Make public and reuse in Search

File: src/main/java/sirius/db/qdrant/QdrantDatabase.java (hunk 3 above)
Comment (andyHa) [reply src/main/java/sirius/db/qdrant/QdrantDatabase.java:40]: We commonly use constants from netty for this - but that's not available in -db - The status is available in HttpUrlConnection...

File: src/main/java/sirius/db/qdrant/QdrantDatabase.java (hunk 4 above)
Comment (andyHa) [reply src/main/java/sirius/db/qdrant/QdrantDatabase.java:60]: not commonly available in -db due to limited dependencies :(

**Review by sabieber — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Pull request merged**